clients/vault/src/metrics.rs

```diff
@@ -497,7 +497,7 @@ pub async fn monitor_bridge_metrics(
 				for currency_id in updated_currencies {
 					for vault in vaults
 						.iter()
-						.filter(|vault| &vault.vault_id.collateral_currency() == currency_id)
+						.filter(|vault| vault.vault_id.collateral_currency() == **currency_id)
```

Comment:
Are two asterisks needed?

Reply:
I think so, because it is a reference to `Static<CurrencyId>`, the second de-reference will get the inner value of the Static wrapper.